src/Doctrine/Odm/State/LinksHandlerTrait.php

```diff
@@ -100,10 +100,6 @@ private function buildAggregation(string $toClass, array $links, array $identifi
 
         $classMetadata = $manager->getClassMetadata($aggregationClass);
 
-        if (!$classMetadata instanceof ClassMetadata) {
```

Comment:
getClassMetadata returns always a ClassMetadata according to the native return type.

Yes, objectManager only has phpdoc for doctrine/persistence < 4, but here we always has a documentManager and this one use native type since 2.0
https://github.com/doctrine/mongodb-odm/blob/cdae04333cc592e3c0d388aa6745d1d486aed6e3/lib/Doctrine/ODM/MongoDB/DocumentManager.php#L282